packages/skill-lesson/trpc/routers/coupons.ts

```diff
@@ -49,7 +49,7 @@ export const couponsRouter = router({
     })
     return coupons
   }),
-  delete: publicProcedure
+  delete: adminProcedure
```

Comment:
This allows us to make something like _delete a coupon_ an admin-only action, where tRPC checks in the middleware if the user is authorized to take the action.